### DISCUSSION TITLE
Handling of FormData

### DISCUSSION THREAD

**Discussion by jasonsaayman** [axios/axios#4474] [Ideas]
Discussion around how we are going to handle from data in axios.

**Comment by DigitalBrainJS** [comment]
> I did not miss that point (see quote at the beginning of your comment 🙂).

@carpben However, the quote consists of two conditions:
- is not in FormData format (there is no such condition ¯\_(ツ)_/¯)
- content-type is multipart/form-data

But there is nothing about plain object type. It doesn't try to detect non FormData type for transformation. Only plain objects, nothing more. Other payloads type just pass through as it was before.

> I thought we need to have a test if a user passes FormData current change doesn't affect it in any way.

FormData is not a plain object type. 
Any object, except: `{}`, `new Object()`, `Object.create(null)` is not a plain object.
That is why that PR could not affect anything, only the case if the user passed a plain object with `multipart/form-data` header to send as JSON string, but this case is erroneous by its nature. It couldn't be used before. So it's should be safe.

**Reply by carpben** [reply]
I was naturally concerned when I noticed all the FormData issue open, and wanted to do things gradual. But I understand your logic, and I think we are good to go.

**Comment by jasonsaayman** [comment]
I like the idea of auto parsing, I also would like the option to keep the old way axios worked. I would like to explore in the new verison of axios this being a module of axios with a default way of working, this would be much better as people would essentially get to choose how they want axios to work with their own setup.

**Comment by ankurk91** [comment]
axios should not convert payload to form-data internally, developer can do it via 3rd party packages,

for example this package
https://www.npmjs.com/package/object-to-formdata

**Comment by jasonsaayman** [comment]
vastly improved form data in the latest version please use that
